src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -244,10 +283,75 @@ public int createParquetFiles(
     sqlCreateTableTemplate.add("ingestFileName", ingestFileName);
     sqlCreateTableTemplate.add("controlFileDataSourceName", controlFileDataSourceName);
     sqlCreateTableTemplate.add("columns", columns);
-
     return executeSynapseQuery(sqlCreateTableTemplate.render());
   }
 
+  public void createSnapshotRowIdsParquetFile(
+      List<DatasetTable> tables,
+      UUID snapshotId,
+      String datasetDataSourceName,
+      String snapshotDataSourceName,
+      String datasetFlightId)
+      throws SQLException {
+
+    // Get all row ids from the dataset
+    List<String> selectStatements = new ArrayList<>();
+    for (DatasetTable table : tables) {
+      String datasetParquetFileName =
+          IngestUtils.getSourceDatasetParquetFilePath(table.getName(), datasetFlightId);
+      ST sqlTableTemplate = new ST(getLiveViewTableTemplate);
+      sqlTableTemplate.add("tableId", table.getId().toString());
+      sqlTableTemplate.add("dataRepoRowId", PDAO_ROW_ID_COLUMN);
+      sqlTableTemplate.add("datasetParquetFileName", datasetParquetFileName);
+      sqlTableTemplate.add("datasetDataSourceName", datasetDataSourceName);
+      selectStatements.add(sqlTableTemplate.render());
```

Comment:
Not a huge deal but `ST.add()` returns itself so it can be chained
```suggestion
      ST sqlTableTemplate = new ST(getLiveViewTableTemplate)
        .add("tableId", table.getId().toString())
        .add("dataRepoRowId", PDAO_ROW_ID_COLUMN)
        .add("datasetParquetFileName", datasetParquetFileName)
        .add("datasetDataSourceName", datasetDataSourceName);
      selectStatements.add(sqlTableTemplate.render());
```